server/events/vcs/bitbucketcloud/client.go

```diff
@@ -124,6 +124,27 @@ func (b *Client) PullIsApproved(repo models.Repo, pull models.PullRequest) (bool
 	return false, nil
 }
 
+// PullIsMergeable returns true if the merge request has no conflicts and can be merged.
+func (b *Client) PullIsMergeable(repo models.Repo, pull models.PullRequest) (bool, error) {
+	// NOTE: The 1.0 API is deprecated, but the 2.0 API does not provide this endpoint.
+	path := fmt.Sprintf("%s/1.0/repositories/%s/pullrequests/%d/conflict-status", b.BaseURL, repo.FullName, pull.Num)
+	resp, err := b.makeRequest("GET", path, nil)
+	if err != nil {
+		return false, err
+	}
+	var conflictStatus ConflictStatus
+	if err := json.Unmarshal(resp, &conflictStatus); err != nil {
+		return false, errors.Wrapf(err, "Could not parse response %q", string(resp))
+	}
+	if err := validator.New().Struct(conflictStatus); err != nil {
+		return false, errors.Wrapf(err, "API response %q was missing fields", string(resp))
+	}
+	if (conflictStatus.MergeImpossible == nil || !*conflictStatus.MergeImpossible) && (conflictStatus.IsConflicted == nil || !*conflictStatus.IsConflicted) {
```

Comment:
You don't need to check nil's here thanks to the validator

Reply:
Hmm, sure about this? Does a JSON `null` get converted to `false`? I'm not sure exactly how the JSON parser works.

Reply:
The parser will set the field to `nil`, however we're performing additional validation via the `validator` library. When you set the struct tags up, it will require those fields to not be `nil`.